### PR TITLE
[NPU] fix qwen3.5 video processor

### PR DESCRIPTION
**Pull request by zhaozx-cn**
## Motivation
In the Qwen3VLVideoProcessor, there is a permute operation with more than 8 dimensions which is not supported on the NPU. Following PR #20189 , this PR applies a patch to the Qwen3VLVideoProcessor in the transformers to avoid processing data with more than 8 dimensions.

## Modifications
Override the _process function in Qwen3VLVideoProcessor to avoid processing of data with more than 8 dimensions.

## Accuracy Tests
```
import requests

url = f"http://localhost:30000/v1/chat/completions"

data = {
    "model": "/home/weights/Qwen3.5-4B",
    "messages": [
        {
            "role": "user",
            "content": [
                {"type": "text", "text": "What’s in this video?"},
                {
                    "type": "video_url",
                    "video_url": {
                        "url": "https://huggingface.co/datasets/raushan-testing-hf/videos-test/resolve/main/sample_demo_1.mp4"
                    },
                },
            ],
        }
    ],
    "max_tokens": 1000,
}

response = requests.post(url, json=data)
print(response.text)
```
NPU

- before this PR:
Execution failed
- after this PR:
_The user wants me to describe the video content.\n\n1.  **Identify the main subject:** A baby with very thin, blonde hair.\n2.  **Identify the accessories:** The baby is wearing dark, thick-rimmed glasses, almost like shades, though it looks like a pair of frames on the head/nose area.\n3.  **Identify the action:** The baby is sitting on a bed or floor, holding a large book. The baby repeatedly turns the pages of the book, flipping them fast. It looks almost like a fan or a drum.\n4.  **Identify the setting:** There's a crib in the background, suggesting a bedroom or nursery. There are clothes scattered around.\n5.  **Synthesize the description:** Combine these elements into a concise summary.\n    *   Baby wearing glasses.\n    *   Sitting on a surface (bed/floor).\n    *   Holding a book.\n    *   Flipping pages rapidly.\n    *   Background context (crib).\n\n**Drafting the response:**\nThe video shows a baby with thin blonde hair and thick-rimmed glasses sitting on a bed (or the floor). The baby is holding a large, open book and is rapidly flipping through the pages, turning them back and forth quite quickly. In the background, there is a wooden crib. The baby seems focused on the book, even while flipping the pages at a high speed.\n\n**Refining:**\nThe core action is \"flipping pages.\" The \"glasses\" are a humorous detail often found in viral videos. The \"book\" looks like a standard children's book, but it's very large for the baby's hands.\n\nLet's stick to the visual facts.\n- Baby with glasses.\n- Flipping pages of a book fast.\n- Sitting down.\n\nFinal check of the prompt: \"What's in this video?\"\nResponse should be a direct description.\n\n\"The video features a baby wearing thick-rimmed glasses sitting on a bed and rapidly flipping through the pages of a large book.\" This covers the main points. I can add detail about the speed. The baby is flipping the pages back and forth very quickly.\n\nLet's formulate the final output.\n</think>\n\nThe video shows a baby with thin hair and thick-rimmed glasses sitting on a bed. The baby is holding a large book and repeatedly flipping through the pages very quickly, almost drumming with the motion. In the background, a wooden crib is visible._

GPU (H20)
_The user wants me to describe the video provided.\n\n**1. Analyze the video content:**\n*   **Subject:** A baby/toddler with very little hair and wearing glasses.\n*   **Clothing:** A light blue sleeveless top and pink pants.\n*   **Action:** The baby is sitting on a bed, holding a book. The baby is flipping through the pages of the book.\n*   **Setting:** A bedroom. There's a bed with a patterned quilt, a wooden crib in the background, and some clothes hanging or piled up.\n*   **Details:** The baby is wearing black-rimmed glasses. The baby seems focused on the book but also a bit messy with flipping pages. There is a \"clideo.com\" watermark in the corner.\n\n**2. Structure the description:**\n*   Start with the main subject and action.\n*   Mention the clothing and accessories (glasses).\n*   Describe the setting.\n*   Add any specific details about the behavior (flipping pages).\n\n**3. Draft the response:**\n*   \"The video shows a tiny baby wearing glasses sitting on a bed.\"\n*   \"The baby is holding a book and flipping through the pages.\"\n*   \"The baby is wearing a light blue shirt and pink pants.\"\n*   \"In the background, there is a crib and a messy bed.\"\n\n**4. Refine the response:**\n*   Combine the sentences for better flow. \"This video features a cute baby wearing black-rimmed glasses sitting on a bed. The baby, dressed in a light blue sleeveless shirt and pink pants, is holding open a book and enthusiastically flipping through the pages. Behind the baby, you can see a wooden crib and other household items like clothes on the bed.\"\n\n**5. Final check against the prompt:** \"What's in this video?\" - The description answers this directly.\n\nLet's write the final output based on these observations.\n- Subject: Baby with glasses.\n- Action: Reading/book flipping.\n- Setting: Bedroom/bed.\n- Specifics: Blue shirt, pink pants.\n\nThe generated response should cover these points clearly.\n</think>\n\nThe video shows a baby wearing glasses sitting on a bed and flipping through the pages of a book. The baby is wearing a light blue sleeveless top and pink pants. In the background, there is a wooden crib and some clothes on the bed. The baby seems quite focused on the book._

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/modules/qwen_vl_processor.py [hunk 1]
```diff
@@ -138,6 +147,138 @@ def _preprocess(
...
+            if pad := -T % temporal_patch_size:
+                repeats = patches[:, -1:].expand(-1, pad, -1, -1, -1)
+                patches = torch.cat((patches, repeats), dim=1)
+            batch_size, grid_t, channel = patches.shape[:3]
+            grid_t = grid_t // temporal_patch_size
+            grid_h, grid_w = resized_height // patch_size, resized_width // patch_size
+
+            ######################################
+            # Start of modifications for sglang  #
+            ######################################
```
Comment (xiaobaicxy) [inline python/sglang/srt/hardware_backend/npu/modules/qwen_vl_processor.py:257]: Please extract a common function
